resources/tag.py

```diff
@@ -39,24 +40,29 @@ def get(self):
                     schema:
                       type: array
                       items:
-                        $ref: '#/components/schemas/Tag'
+                        $ref: '#/components/schemas/TagItemResponse'
         """
-        tag_prefix = request.args.get('query')
-
-        if not tag_prefix or len(tag_prefix) == 0:
-            tags = []
-        else:
-            tag_prefix = tag_prefix.lower()
-            tags = db.session.query(Tag.tag)\
-                             .distinct(Tag.tag)\
-                             .join(object_tag_table, object_tag_table.c.tag_id == Tag.id) \
-                             .join(ObjectPermission,
-                                   and_(ObjectPermission.object_id == object_tag_table.c.object_id,
-                                        g.auth_user.is_member(ObjectPermission.group_id))) \
-                             .filter(Tag.tag.like(tag_prefix + "%")).all()
-        multi_tag = TagSchema(many=True)
-        dumped_tags = multi_tag.dump(tags)
-        return dumped_tags
+        schema = TagListRequestSchema()
+        obj = schema.load(request.args)
+        if obj.errors:
+            return {"errors": obj.errors}, 400
+
+        tags = (
+            db.session.query(Tag.tag)
+                      .distinct(Tag.tag)
+                      .join(object_tag_table, object_tag_table.c.tag_id == Tag.id)
+                      .join(ObjectPermission,
+                            and_(ObjectPermission.object_id == object_tag_table.c.object_id,
+                                 g.auth_user.is_member(ObjectPermission.group_id)))
+        )
+        tag_prefix = obj.data["query"]
+
+        if tag_prefix:
```

Comment:
```suggestion

        tag_prefix = obj.data["query"]
        if tag_prefix:
```
stylistic change, but implies that `tag_prefix` is used in the 2nd block, not first